releasetool/commands/start/go.py

```diff
@@ -0,0 +1,206 @@
+    changes: Sequence[str] = ()
+    release_notes: Optional[str] = None
+    last_release_version: Optional[str] = None
+    last_release_committish: Optional[str] = None
+    release_version: Optional[str] = None
+    release_branch: Optional[str] = None
+#    pull_request: Optional[dict] = None
+
+
+def determine_module_name(ctx: Context) -> None:
+    # Get the module name from the go.mod file in the current directory.
+    click.secho("> Figuring out the module name.", fg="cyan")
+    info = read_gomod()
+    if info is None:
+        click.secho("Looks like we're releasing the repo (no modules).")
+    else:
+        ctx.module_name = info["Module"]["Path"]
+        ctx.relative_module_name = relative_module_name(ctx.module_name)
+        click.secho(f"Looks like we're releasing {ctx.module_name} (relative path {ctx.relative_module_name}).")
+
+def read_gomod() -> Optional[Dict[str, str]]:
+    if os.path.isfile("go.mod"):
+        output = subprocess.check_output(["go", "mod", "edit", "-json"]).decode("utf-8")
+        return json.loads(output)
+    elif os.path.isdir(".git"):
+        # No go.mod; must release from repo root.
+        return None
+
+# Returns modname relative to the repo root, assuming modname's go.mod file is
+# in the current directory.
+def relative_module_name(modname) -> int:
+    dir = os.getcwd()
+    components = []
+    while dir != "/":
+        if os.path.isdir(os.path.join(dir, ".git")):
+            return '/'.join(reversed(components))
+        dir, c = os.path.split(dir)
+        components.append(c)
+    raise ValueError("not inside a git repo")
+
+def determine_last_release(ctx: Context) -> None:
+    click.secho("> Figuring out what the last release was.", fg="cyan")
+    if ctx.relative_module_name is None:
+        prefix = "v"
+    else:
+        prefix = ctx.relative_module_name + "/v"
+    tags = releasetool.git.list_tags()
+    candidates = [
+        tag
+        for tag in tags
+        if tag.startswith(prefix)
+    ]
+
+    if candidates:
+        ctx.last_release_committish = candidates[0]
+        ctx.last_release_version = candidates[0].rsplit("/").pop()[1:]
+
+    else:
+        click.secho(
+            f"I couldn't figure out the last release for {ctx.module_name}, "
+            "so I'm assuming this is the first release. Can you tell me "
+            "which git rev/sha to start the changelog at?",
+            fg="yellow",
+        )
+        ctx.last_release_committish = click.prompt("Committish")
+        ctx.last_release_version = "0.0.0"
+
+    click.secho(f"The last release was {ctx.last_release_version}.")
+
+
+def gather_changes(ctx: Context) -> None:
+    click.secho(f"> Gathering changes since {ctx.last_release_version}", fg="cyan")
+    ctx.changes = releasetool.git.summary_log(
+        from_=ctx.last_release_committish, to=f"origin/master"
+    )
+    click.secho(f"Cool, {len(ctx.changes)} changes found.")
+
+
+def determine_release_version(ctx: Context) -> None:
```

Comment:
@busunkim96 did we ever pull this out into `common`?

Reply:
No, just `edit_release_notes` I'll make a PR to put this in `common` as well.